haskell/private/haskell_impl.bzl

```diff
@@ -401,33 +408,36 @@ def haskell_library_impl(ctx):
     exposed_modules_file = list_exposed_modules(
         hs,
         ls_modules = ctx.executable._ls_modules,
+        modules = modules,
         other_modules = other_modules,
         exposed_modules_reexports = exposed_modules_reexports,
         interfaces_dir = c.interfaces_dir,
         with_profiling = with_profiling,
     )
 
-    if srcs_files:
+    if srcs_files or modules:
```

Comment:
Perhaps this could be factored out into a variable, `non_empty` or something of that sort.

Reply:
Yeah, I was debating about this and don't remember why I didn't end up doing it. Given that I can't think of an objection now, I'll assume there wasn't one before either.